### PR TITLE
Chapter1-Homework3

### PR DESCRIPTION
**Pull request by davinsoman**
I hope I was able to complete this to the requirements and pass the checks/test :)

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            return int.Parse(Console.ReadLine());
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return float.Parse(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float heightM)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 1
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            return int.Parse(Console.ReadLine());
+        }
+
+        public static float PromptFloat(string message)
+        {
+            Console.WriteLine(message);
+            return float.Parse(Console.ReadLine());
+        }
+
+        public static float CalculateBMI(float weight, float height)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Lesson3.cs]: Codacy Issue found: [Rename method 'CalculateBMI' to match pascal case naming rules, consider using 'CalculateBmi'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -17,25 +17,25 @@ public static class Checks
         public static int PromptInt(string message)
         {
             // To do: call your implementation. 
-            return 0;
+            return Lesson3.PromptInt(message);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Checks.cs]: Please remove the to do comments when you have implemented them. :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            Console.WriteLine("Please enter your data" + Environment.NewLine);
+            do
+            {
+                var name = PromptString("What is your first name?");
+                var surname = PromptString("What is your last name?");
+                var age = PromptInt("What is your age");
+                var weight = PromptFloat("What is your weight in kilograms");
+                var heightCm = PromptFloat("What is your height in centimeters");
+                var BMI = CalculateBmi(weight, heightCm / 100);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Your variable name should be bmi here.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+namespace BootCamp.Chapter
+{
+    public class Lesson3
+    {
+        public static void Demo()
+        {
+            Console.WriteLine("Please enter your data" + Environment.NewLine);
+            do
+            {
+                var name = PromptString("What is your first name?");
```
Comment (diverges) [inline Src/BootCamp.Chapter/Lesson3.cs]: This can probably be its own function :)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 6]
```diff
@@ -15,14 +15,14 @@ public static void Demo()
                 var age = PromptInt("What is your age");
                 var weight = PromptFloat("What is your weight in kilograms");
                 var heightCm = PromptFloat("What is your height in centimeters");
-                var BMI = CalculateBmi(weight, heightCm / 100);
+                var Bmi = CalculateBmi(weight, heightCm / 100);
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/Lesson3.cs]: Being a bit picky here but keeping up with your own style is very important if you advance in programming and your projects get larger and you end up with multiple files that you want to read quickly without getting disturbed by different styles.
Change it to be all lowercase. Your local variables should all start with a lowercase letter and should only have one uppercase letter for each new word beginning in the variable name. :)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4863985&bid=16132144)

**Pull request closed without merging**